pgactivity/views.py

```diff
@@ -199,55 +229,105 @@ def header(
                 f"{term.bold}{host.hostname}{term.normal}",
                 f"{term.cyan}{pg_host}{term.normal}",
                 f"Ref.: {ui.refresh_time}s",
+                f"Duration mode: {ui.duration_mode.name}",
             ]
             + ([f"Min. duration: {ui.min_duration}s"] if ui.min_duration else [])
         )
     )
 
-    total_size = utils.naturalsize(dbinfo.total_size)
-    size_ev = utils.naturalsize(dbinfo.size_ev)
-
-    def render_columns(columns: List[List[str]], *, delimiter: str) -> Iterator[str]:
-        column_widths = [
-            max(len(column_row) for column_row in column) for column in columns
+    total_size = utils.naturalsize(si.total_size)
```

Comment:
Should all thoses natural size calls be moved to the server_informations class ? (generate a total_size_natural or add a method that does this in the class ? or an methode disguised as an attribute)

Or should I avoid this and keep a distinction between the data and it's visualization ?

Reply:
Yes, that's fine as it is now; keep this logic at rendering time.

Reply:
Sorry for asking again, I want to make sure I understand the rationale.
Is it better to :

- keep the logic in different places ?
- keep the execution of the logic during rendering time ? in that case, would a property do the job ?

Reply:
It is better to keep data untransformed in the `ServerInformation` object and then format it here in the view. I don't think a property is worth because it'll be only used once.

The logic is:
* get data from the database
* possibly transform it into something that makes sense for our application (rounding does not count as such, but statistics would *if* we don't store the underlying data -- for instance, you don't store the mean value of a series along with the series itself, otherwise the statistics may get inconsistent with actual data)
* store raw data in a live object
* render data from that object in a view, where "render" can mean rounding, simple calculation, etc.

Reply:
thx for the explanation.